Sources/Architecture/ARM/SoftwareBreakpointManager.cpp

```diff
@@ -35,7 +35,10 @@ void SoftwareBreakpointManager::clear() {
 }
 
 ErrorCode SoftwareBreakpointManager::add(Address const &address, Type type,
-                                         size_t size) {
+                                         size_t size, Mode mode) {
+  if (mode != kModeExec)
```

Comment:
I think you can make this a `DS2ASSERT`, given that we don't pass user data directly here, and that all callers of `::add` will always call it with `kModeExec`.